_includes/head.html

```diff
@@ -1,8 +1,42 @@
+{%- comment -%}
+  Include as: {%- include head.html -%}
+  Depends on: site.ga_tracking, site.ga_tracking_anonymize_ip,
+    site.search_enabled, site.static_files, site.favicon_ico.
+  Results in: HTML for the head element.
+  Includes:
+    css/activation.scss.liquid, head_custom.html.
+  Overwrites: 
+    scss, ga_tracking_ids, ga_property, file, favicon.
+  Should not be cached, because css/activation.scss.liquid depends on page.
+{%- endcomment -%}
+
 <head>
   <meta charset="UTF-8">
   <meta http-equiv="X-UA-Compatible" content="IE=Edge">
 
   <link rel="stylesheet" href="{{ '/assets/css/just-the-docs-default.css' | relative_url }}">
+  
+  <style type="text/css">
+    {%- capture scss -%}
+      @import "./support/support";
+      @import "./custom/setup";
+      {%- if site.color_scheme and site.color_scheme != "nil" -%}
+        {%- assign color_scheme = site.color_scheme -%}
+      {%- else -%}
+        {%- assign color_scheme = "light" -%}
+      {%- endif -%}
+      @import "./color_schemes/light";
+      {% unless color_scheme == "light" %}
+      @import "./color_schemes/{{ color_scheme }}";
+      {% endunless %}
+      {%- include css/activation.scss.liquid -%}
+    {%- endcapture -%}
+    {%- comment -%}
+      Convert to CSS, then remove the color_scheme import rules to avoid duplication
+      (assuming that scssify of the color_scheme doesn't contain `.site-nav`):
+    {%- endcomment %}
+    {{ scss | scssify | split: ".site-nav" | shift | join: ".site-nav" | prepend: ".site-nav" }}
```

Comment:
Is this something we should make more clear to users?

Reply:
It's probably worth a footnote in [the docs](https://just-the-docs.github.io/just-the-docs/docs/customization/#define-a-custom-scheme).

I _think_ `".site-nav"` could be replaced by `".site-nav > .nav-list:nth-"`, but I'm not sure that `scssify` always preserves how selectors are written. 

It seems the duplication of the imported CSS would disappear (along with the need to match the end of it) if we could [change `@import` to `@use` in the SCSS](https://sass-lang.com/documentation/at-rules/use/#differences-from-import). But that is surely out of scope for this PR.

Reply:
I think a footnote is alright for now; agreed that changing to `@use` is out of scope for this PR.

Reply:
@mattxwang I've now eliminated the constraint about not using `.site-nav` in custom color schemes.

It required some extra Liquid code, complicating `head.html`, so I've now moved all the Liquid code to a new include: `head_nav.html`.